koji/collection-mash-split.py

```diff
@@ -160,6 +167,32 @@ def createrepo(self, path, comps=None, checksum=None):
         self.logger.debug("running %s" % cmd)
         kobo.shortcuts.run(cmd)
 
+    def inject_modulemd_yml(self, path, modulemd_yaml):
+        """Generate modular metadata and inject it.
+
+        @param path: path to repository
+        @type path: str
+        @param modulemd_yaml: metadata template
+        @type modulemd_yaml: str
+        """
+        # The epoch number in the NEVRA string is mandatory per spec
+        # https://github.com/fedora-modularity/libmodulemd/blob/main/yaml_specs/modulemd_stream_v2.yaml#L668
+        # So we can't just use `%{nevra}` here
+        cmd = "rpm --query --package {}/*rpm".format(path)
+        cmd += " --queryformat='%{name}-%{epochnum}:%{version}-%{release}.%{arch}\n'"
+        self.logger.debug("running %s" % cmd)
+        status, output = kobo.shortcuts.run(cmd)
+        with open(modulemd_yaml) as modulemd_file:
+            modules = yaml.safe_load(modulemd_file)
+        modules['data']['artifacts'] = {'rpms': output.split()}
```

Comment:
Should it split on lines, in case there's a space in a filename somewhere?

Reply:
strictly speaking, those aren't file names, but NEVRAs, which shouldn't contain spaces.

but better safe than sorry, I guess.